File: master_of_plow_CPP_SDK/src-gen/nevonex/fcal/AGMO_GENERIC_SEND_CMD_T_W.cpp
/*
 * nevonex/fcal/AGMO_GENERIC_SEND_CMD_T_W.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "AGMO_GENERIC_SEND_CMD_T_W.hpp"
#include <ecore/EObject.hpp>
#include <ecore/EClass.hpp>
#include "nevonex/fcal/FcalPackage.hpp"
#include <ecorecpp/mapping.hpp>
#include <algorithm>
#include <nevonex/fcb/SubscriberEnum.hpp>
#include <nevonex-fcal-platform/log/Logger.hpp>

#if BOOST_VERSION >= 106501
#define BOOST_STACKTRACE_HEADERS_FOUND
#include <boost/stacktrace.hpp>
#endif

/*PROTECTED REGION ID(AGMO_GENERIC_SEND_CMD_T_W.cpp) START*/
// Please, enable the protected region if you add manually written code.
// To do this, add the keyword ENABLED before START.
/*PROTECTED REGION END*/

using namespace ::nevonex::fcal;
using namespace ::nevonex::log;

// Default constructor
AGMO_GENERIC_SEND_CMD_T_W::AGMO_GENERIC_SEND_CMD_T_W() : m_aD_Generic_Cmd(0), m_aD_Generic_Mode_Cmd(0)
{

    /*PROTECTED REGION ID(AGMO_GENERIC_SEND_CMD_T_W__AGMO_GENERIC_SEND_CMD_T_W) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

AGMO_GENERIC_SEND_CMD_T_W::~AGMO_GENERIC_SEND_CMD_T_W()
{

    /*PROTECTED REGION ID(AGMO_GENERIC_SEND_CMD_T_W__Destructor) START*/
    // Please, enable the protected region if you add manually written code.
    // To do this, add the keyword ENABLED before START.
    /*PROTECTED REGION END*/

}

void AGMO_GENERIC_SEND_CMD_T_W::initialize(::ecore::EInt _aD_Generic_Cmd, ::ecore::EInt _aD_Generic_Mode_Cmd)
{
    m_aD_Generic_Cmd = _aD_Generic_Cmd;
    m_aD_Generic_Mode_Cmd = _aD_Generic_Mode_Cmd;
}

// Attributes

::ecore::EInt AGMO_GENERIC_SEND_CMD_T_W::getAD_Generic_Cmd() const
{
    return m_aD_Generic_Cmd;
}

void AGMO_GENERIC_SEND_CMD_T_W::setAD_Generic_Cmd(::ecore::EInt _aD_Generic_Cmd)
{

    ::ecore::EInt _old_aD_Generic_Cmd = m_aD_Generic_Cmd;
    m_aD_Generic_Cmd = _aD_Generic_Cmd;

}

::ecore::EInt AGMO_GENERIC_SEND_CMD_T_W::getAD_Generic_Mode_Cmd() const
{
    return m_aD_Generic_Mode_Cmd;
}

void AGMO_GENERIC_SEND_CMD_T_W::setAD_Generic_Mode_Cmd(::ecore::EInt _aD_Generic_Mode_Cmd)
{

    ::ecore::EInt _old_aD_Generic_Mode_Cmd = m_aD_Generic_Mode_Cmd;
    m_aD_Generic_Mode_Cmd = _aD_Generic_Mode_Cmd;

}

// References

File: com.bosch.fsp.master_of_plow.gen/src/com/bosch/nevonex/fcal/impl/AGMO_GENERIC_SEND_CMD_T_W.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.fcal.impl;

import com.bosch.nevonex.fcal.IAGMO_GENERIC_SEND_CMD_T_W;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.EObject;
import org.eclipse.emf.ecore.InternalEObject;

import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * @generated
 */
public class AGMO_GENERIC_SEND_CMD_T_W extends EObjectImpl implements IAGMO_GENERIC_SEND_CMD_T_W {
	protected int aD_Generic_Cmd = 0;
	protected int aD_Generic_Mode_Cmd = 0;

	protected AGMO_GENERIC_SEND_CMD_T_W() {
		super();
	}

	@Override
	protected EClass eStaticClass() {
		return FcalPackage.Literals.AGMO_GENERIC_SEND_CMD_T_W;
	}

	public int getAD_Generic_Cmd() {
		return aD_Generic_Cmd;
	}

	public void setAD_Generic_Cmd(int newAD_Generic_Cmd) {
		aD_Generic_Cmd = newAD_Generic_Cmd;
	}

	public int getAD_Generic_Mode_Cmd() {
		return aD_Generic_Mode_Cmd;
	}

	public void setAD_Generic_Mode_Cmd(int newAD_Generic_Mode_Cmd) {
		aD_Generic_Mode_Cmd = newAD_Generic_Mode_Cmd;
	}

	/**
	 * @generated
	 */
	@Override
	public void initialize(int aD_Generic_Cmd, int aD_Generic_Mode_Cmd) {
		this.aD_Generic_Cmd = aD_Generic_Cmd;
		this.aD_Generic_Mode_Cmd = aD_Generic_Mode_Cmd;
	}

	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
		case FcalPackage.AGMO_GENERIC_SEND_CMD_T_W__AD_GENERIC_CMD:
			return getAD_Generic_Cmd();
		case FcalPackage.AGMO_GENERIC_SEND_CMD_T_W__AD_GENERIC_MODE_CMD:
			return getAD_Generic_Mode_Cmd();
		}
		return super.eGet(featureID, resolve, coreType);
	}

	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
		case FcalPackage.AGMO_GENERIC_SEND_CMD_T_W__AD_GENERIC_CMD:
			setAD_Generic_Cmd((Integer) newValue);
			return;
		case FcalPackage.AGMO_GENERIC_SEND_CMD_T_W__AD_GENERIC_MODE_CMD:
			setAD_Generic_Mode_Cmd((Integer) newValue);
			return;
		}
		super.eSet(featureID, newValue);
	}

	@Override
	public void eUnset(int featureID) {
		switch (featureID) {
		case FcalPackage.AGMO_GENERIC_SEND_CMD_T_W__AD_GENERIC_CMD:
			setAD_Generic_Cmd(0);
			return;
		case FcalPackage.AGMO_GENERIC_SEND_CMD_T_W__AD_GENERIC_MODE_CMD:
			setAD_Generic_Mode_Cmd(0);
			return;
		}
		super.eUnset(featureID);
	}

	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
		case FcalPackage.AGMO_GENERIC_SEND_CMD_T_W__AD_GENERIC_CMD:
			return aD_Generic_Cmd != 0;
		case FcalPackage.AGMO_GENERIC_SEND_CMD_T_W__AD_GENERIC_MODE_CMD:
			return aD_Generic_Mode_Cmd != 0;
		}
		return super.eIsSet(featureID);
	}
} //AGMO_GENERIC_SEND_CMD_T_W
